### PR TITLE
[RLlib] MADDPG: Move into main `algorithms` folder and add proper unit and learning tests.

### PR DESCRIPTION
**Pull request by sven1977**
MADDPG: Move into main `algorithms` folder and transfer to `training_iteration` API.
* Add better unit tests for this algo.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/BUILD [hunk 1]
```diff
@@ -379,6 +379,17 @@ py_test(
     args = ["--yaml-dir=tuned_examples/impala"]
 )
 
+# MADDPG
+py_test(
+    name = "learning_tests_two_step_game_maddpg",
+    main = "tests/run_regression_tests.py",
+    tags = ["team:ml", "tf_only", "no_tf_eager_tracing", "learning_tests", "learning_tests_discrete"],
```
Comment (avnishn) [inline rllib/BUILD:386]: Could we maybe instead add this to the weekly ci -- I feel like we don't need to as widely support this algorithm until there is a customer use case.

**Review by avnishn — APPROVED** [review]
Does this learning test have to be run on pull requests? It feels like it doesn't impact our critical users, and can be moved to weekly.

File: rllib/BUILD (hunk 1 above)
Comment (sven1977) [reply rllib/BUILD:386]: Good question: I think we should move toward:
* CI-tests: small tasks that can be learnt in 1-2 min (CartPole, **TwoStepGame**, maybe Pendulum or simpler cont. actions problem)
* nightly tests: hard tasks, requiring GPU + ~1h of completion on >1 workers

Comment (sven1977) [reply rllib/BUILD:386]: I'll leave this test here for now to roughly match the other algos' coverage (everyone has CartPole/Pendulum tests in the CI, which is ok).

**Pull request merged**